lib/auth/saml.go

```diff
@@ -31,7 +30,11 @@ import (
 
 // ErrSAMLRequiresEnterprise is the error returned by the SAML methods when not
 // using the Enterprise edition of Teleport.
-var ErrSAMLRequiresEnterprise = fmt.Errorf("SAML: %w", ErrRequiresEnterprise)
+//
+// TODO(zmb3): ideally we would wrap ErrRequiresEnterprise here, but
+// we can't currently propagate wrapped errors across the gRPC boundary,
+// and we want tctl to display a clean user-facing messasge in this case
```

Comment:
Nit: This comment feels out of place on this variable declaration tbh, or maybe I misunderstood.

Reply:
I don't want someone to accidentally change this, because `trace` is very finicky with gRPC. If you wrap the error in any way, the user-facing error message that tctl prints will have extra RPC garbage in it.